packages/legend-query-builder/src/components/QueryLoader.tsx

```diff
@@ -303,136 +342,203 @@ export const QueryLoader = observer(
+                                >
+                                  Me
+                                </div>
+                              ) : (
+                                query.owner
+                              )}
+                            </div>
                           </div>
-                        ) : (
+                        </div>
+                        <DropdownMenu
+                          className="query-loader__result__actions-menu"
+                          title="More Actions..."
+                          content={
+                            <MenuContent>
+                              <MenuContentItem
+                                onClick={(): void => showPreview(query.id)}
+                              >
+                                Show Query Preview
+                              </MenuContentItem>
+                              {!queryLoaderState.isReadOnly && (
+                                <MenuContentItem
+                                  disabled={!query.isCurrentUserQuery}
+                                  onClick={deleteQuery(query)}
+                                >
+                                  Delete
+                                </MenuContentItem>
+                              )}
+                              {!queryLoaderState.isReadOnly && (
+                                <MenuContentItem
+                                  disabled={!query.isCurrentUserQuery}
+                                  onClick={showEditQueryNameInput(
+                                    query.name,
+                                    idx,
+                                  )}
+                                >
+                                  Rename
+                                </MenuContentItem>
+                              )}
+                            </MenuContent>
+                          }
+                          menuProps={{
+                            anchorOrigin: {
+                              vertical: 'bottom',
+                              horizontal: 'left',
+                            },
+                            transformOrigin: {
+                              vertical: 'top',
+                              horizontal: 'left',
+                            },
+                            elevation: 7,
+                          }}
+                        >
+                          <MoreVerticalIcon />
+                        </DropdownMenu>
+                        <div className="query-loader__result__arrow">
+                          <ThinChevronRightIcon />
+                        </div>
+                      </div>
+                    ))}
+                {queryLoaderState.queryBuilderState &&
+                  queryLoaderState.isCuratedTemplateToggled &&
+                  curatedTemplateQueries
+                    .slice(0, QUERY_LOADER_TYPEAHEAD_SEARCH_LIMIT)
```

Comment:
The current condition is that 
since only `DSL_DataSpace_LegendApplicationPlugin` will return a list of template queries therefore query builder state must be an instance of datapace builder state. Besides, we only display template queries under current dataspace